android/src/main/java/com/yubico/yubikit/android/transport/usb/FidoDeviceFilter.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2025 Yubico.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yubico.yubikit.android.transport.usb;
+
+import android.hardware.usb.UsbConstants;
+import android.hardware.usb.UsbDevice;
+import android.hardware.usb.UsbDeviceConnection;
+import android.hardware.usb.UsbInterface;
+import android.hardware.usb.UsbManager;
+import java.util.Arrays;
+import org.jspecify.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A filter for identifying USB devices that support the FIDO HID protocol. This class matches all
+ * vendor and product IDs, and checks device properties to determine FIDO HID compatibility using
+ * the HID usage page.
+ */
+public class FidoDeviceFilter extends DeviceFilter {
+  private static final byte[] fidoUsagePage = new byte[] {0x06, (byte) 0xD0, (byte) 0xF1};
+  private final Logger logger = LoggerFactory.getLogger(FidoDeviceFilter.class);
+
+  /**
+   * Matches any vendor and product ID combination.
+   *
+   * @param vendorId the USB vendor ID
+   * @param productId the USB product ID
+   * @return always true, as all IDs are accepted
+   */
+  @Override
+  public boolean matchesVendorProduct(int vendorId, int productId) {
+    // match all vendors and products
+    return true;
+  }
+
+  /**
+   * Checks if the specified USB device supports the FIDO HID usage page.
+   *
+   * @param manager the UsbManager instance
+   * @param usbDevice the UsbDevice to check
+   * @return true if the device matches FIDO HID criteria, false otherwise
+   */
+  @Override
+  public boolean canUseDevice(UsbManager manager, UsbDevice usbDevice) {
+    return isFidoDevice(manager, usbDevice);
+  }
```

Comment:
Why so many methods that only call another method?

I would suggest to make the constants in hasFidoUsagePage real constants (static final) in the class, and just have the logic directly in canUseDevice.